flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -255,14 +258,24 @@ protected JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoad
 		}
 	}
 
-	/**
-	 * Requests the {@link JobResult} for the given {@link JobID}. The method retries multiple
-	 * times to poll the {@link JobResult} before giving up.
-	 *
-	 * @param jobId specifying the job for which to retrieve the {@link JobResult}
-	 * @return Future which is completed with the {@link JobResult} once the job has completed or
-	 * with a failure if the {@link JobResult} could not be retrieved.
-	 */
+	@Override
+	public CompletableFuture<JobStatus> getJobStatus(JobID jobId) {
+		JobDetailsHeaders detailsHeaders = JobDetailsHeaders.getInstance();
+		final JobMessageParameters  params = new JobMessageParameters();
+		params.jobPathParameter.resolve(jobId);
+
+		CompletableFuture<JobDetailsInfo> responseFuture = sendRequest(detailsHeaders, params);
+
+		return responseFuture.thenApply((JobDetailsInfo jobDetailsInfo) -> {
+			if (jobDetailsInfo != null) {
+				return jobDetailsInfo.getJobStatus();
+			}
+
+			throw new RuntimeException("Unknown JobStatus.");
```

Comment:
The alternative I had in mind was to extend `JobStatus` with `UNKNOWN`. WDYT?

Reply:
OK, I'm simplifying to

```
responseFuture.thenApply(JobDetailsInfo::getJobStatus);
```

which will let the error response go out in case the job wasn't found.